packages/util/src/array.ts

```diff
@@ -49,3 +49,29 @@ export function splitInChunks<T>(arr: T[], chunkSize: number): T[][] {
   }
   return result;
 }
+
+/**
+ * We want to provide methods which accept a variable single number of elements and arrays.
+ * The overloaded signature to achieve this is:
+ * function doSomething(array:T[])
+ * function doSomething(...varArgs:T[])
+ * functiondoSomething(first:undefined|T|T[],...rest:T[]){
+ *   //implementation
+ * }
+ * This wrapper methods makes it easy build an array from the input.
+ * @param firstOrArray - Either an array, the first element of the var args or undefined if no argument was given.
+ * @param rest - Second to last element if var args were used, empty array if the frist argument is an array.
+ * @returns Array from the input or empty array if no input was given.
+ */
+export function variableArgumentToArray<T>(
+  firstOrArray: undefined | T | T[],
+  rest: T[]
+): T[] {
+  if (Array.isArray(firstOrArray)) {
+    return [...firstOrArray, ...rest];
+  }
+  return firstOrArray ? [firstOrArray, ...rest] : [...rest];
+}
+
+type Rest<T> = T extends T[] ? undefined : T[];
+type First<T> = undefined | T | T[];
```

Comment:
Are these types used anywhere?

Reply:
no I tried something there but my type kung fu was not strong enough and it would also not more understandable.